File: api-frontend/apps/web-antd/src/api/system/site.ts
import { requestClient } from '#/api/request';

export interface SiteRecord {
  id: number;
  tenant_id: number;
  name: string;
  type: string;
  url: string;
  image: string;
  is_domain: number;
  status: number;
  sort: number;
  remark: string;
  created_at: string;
}

export function getSiteList(keyword?: string) {
  return requestClient.get<SiteRecord[]>('/sites', { params: { keyword } });
}

export function createSite(data: Partial<SiteRecord>) {
  return requestClient.post('/sites', data);
}

export function updateSite(id: number, data: Partial<SiteRecord>) {
  return requestClient.put(`/sites/${id}`, data);
}

export function deleteSite(id: number) {
  return requestClient.delete(`/sites/${id}`);
}

File: api-backend/internal/model/site.go
package model

type Site struct {
	TenantBaseModel
	Name     string `gorm:"size:200;not null" json:"name"`
	Type     string `gorm:"size:50" json:"type"`
	URL      string `gorm:"size:500" json:"url"`
	Image    string `gorm:"size:500" json:"image"`
	IsDomain int8   `gorm:"default:0" json:"is_domain"`
	Status   int8   `gorm:"default:1" json:"status"`
	Sort     int    `gorm:"default:0" json:"sort"`
	Remark   string `gorm:"size:500" json:"remark"`
}
